modules/gdscript/gdscript_tokenizer.cpp

```diff
@@ -359,6 +359,11 @@ GDScriptTokenizer::Token GDScriptTokenizerText::make_token(Token::Type p_type) {
 	token.rightmost_column = rightmost_column;
 	token.source = String(_start, _current - _start);
 
+	if (p_type == Token::UNDERSCORE) {
+		// UNDERSCORE could be part of an identifier.
+		token.literal = "_";
+	}
```

Comment:
You could do it here instead:

https://github.com/godotengine/godot/blob/97b8ad1af0f2b4a216f6f1263bef4fbc69e56c7b/modules/gdscript/gdscript_tokenizer.cpp#L575-L578

This would be a bit more optimized as it would remove branching from the general case.

Reply:
Changed it. The original idea was to prevent similar issues if someone ever was to construct an underscore token in another place, but since there is a testcase we should be fine.